server/tests/api/videos/video-channel-syncs.ts

```diff
@@ -0,0 +1,231 @@
+import { FIXTURE_URLS } from '@server/tests/shared'
+import { areHttpImportTestsDisabled } from '@shared/core-utils'
+import { HttpStatusCode, VideoChannelSyncState, VideoInclude, VideoPrivacy } from '@shared/models'
+import {
+  ChannelSyncsCommand,
+  createSingleServer,
+  PeerTubeServer,
+  setAccessTokensToServers,
+  setDefaultVideoChannel,
+  waitJobs
+} from '@shared/server-commands'
+import { expect } from 'chai'
+import 'mocha'
+
+describe('Test channel synchronizations', function () {
+  if (areHttpImportTestsDisabled()) return
+
+  function runSuite (mode: 'youtube-dl' | 'yt-dlp') {
+    describe('Sync using ' + mode, function () {
+      let server: PeerTubeServer
+      let command: ChannelSyncsCommand
+      let startTestDate: Date
+      const userInfo = {
+        accessToken: '',
+        username: 'user1',
+        get channelName () {
+          return this.username + "_channel"
+        },
+        channelId: -1,
+        syncId: -1
+      }
+
+      async function changeCreationDateForSync (channelSyncId: number, creationDate: string) {
+        await server.sql.updateQuery(`
+            UPDATE "videoChannelSync"
+            SET "createdAt"='${creationDate}'
+            WHERE id=${channelSyncId}
+          `)
+      }
+
+      before(async function () {
+        this.timeout(120_000)
+        startTestDate = new Date()
+        server = await createSingleServer(1, {
+          import: {
+            videos: {
+              http: {
+                youtube_dl_release: {
+                  url: mode === 'youtube-dl'
+                    ? 'https://yt-dl.org/downloads/latest/youtube-dl'
+                    : 'https://api.github.com/repos/yt-dlp/yt-dlp/releases',
+
+                  name: mode
+                }
+              }
+            }
+          }
+        })
+        await setAccessTokensToServers([ server ])
+        await setDefaultVideoChannel([ server ])
```

Comment:
Also use `setDefaultChannelAvatar` and `setDefaultAccountAvatar` to ensure we don't have bad `total` results from REST API